python/ray/serve/api.py

```diff
@@ -672,35 +672,33 @@ def connect() -> Client:
     return Client(controller, controller_name, detached=True)
 
 
-def get_current_backend_tag() -> BackendTag:
-    """When called from within a backend, return its backend tag.
+@dataclass
+class BackendContext:
+    """Stores data for Serve API calls from within the user's backend code."""
+    backend_tag: BackendTag
+    replica_tag: ReplicaTag
 
-    Raises:
-        RayServeException if not called from within a Ray Serve backend.
-    """
-    if _INTERNAL_REPLICA_CONTEXT is None:
-        raise RayServeException("`serve.get_current_backend_tag()`"
-                                "may only be called from within a"
-                                "Ray Serve backend.")
-    else:
-        return _INTERNAL_REPLICA_CONTEXT.backend_tag
 
+def get_backend_context():
+    """When called from a backend, return the backend tag and replica tag.
 
-def get_current_replica_tag() -> ReplicaTag:
-    """When called from within a backend, return its replica tag.
+    A replica tag uniquely identifies a single replica for a Ray Serve
+    backend.  Replica tags are of the form `<backend tag>#<random letters>`.
 
-    A replica tag uniquely identifies a single replica (a process)
-    for a Ray Serve backend.
+    Example:
 
-    Raises:
-        RayServeException if not called from within a Ray Serve backend.
+        >>> serve.get_backend_context().backend_tag # my_backend
+        >>> serve.get_backend_context().replica_tag # my_backend#krcwoa
     """
     if _INTERNAL_REPLICA_CONTEXT is None:
-        raise RayServeException("`serve.get_current_replica_tag()`"
-                                "may only be called from within a"
+        raise RayServeException("`serve.get_backend_context()` "
+                                "may only be called from within a "
                                 "Ray Serve backend.")
     else:
-        return _INTERNAL_REPLICA_CONTEXT.replica_tag
+        backend_tag = _INTERNAL_REPLICA_CONTEXT.backend_tag
+        replica_tag = _INTERNAL_REPLICA_CONTEXT.replica_tag
+
+        return BackendContext(backend_tag, replica_tag)
```

Comment:
can we just merge InternalReplicaContext and BackendContext? You can make the `controller_name` a private field `_internal_controller_name` or something. So we don't have an extra data structure that's just a subset of InternalReplicaContext. 

additionally, naming wise,  this is ReplicaContext, not BackendContext because it contains the replica_tag.

Reply:
Yup, that makes a lot of sense